app/models/pseud.rb

```diff
@@ -390,9 +391,19 @@ def check_default_pseud
   end
 
   def expire_caches
-    if saved_change_to_name?
-      works.touch_all
-      series.each(&:expire_byline_cache)
+    return unless saved_change_to_name? || (destroyed? && !user.nil?)
+
+    pseud = destroyed? ? user.default_pseud : self
+    return if pseud.nil?
+
+    pseud.series.each(&:expire_byline_cache)
+    pseud.works.each do |work|
+      work.touch
+      work.expire_caches
+    end
+    pseud.gift_works.each do |work|
+      work.touch
+      work.expire_caches
     end
```

Comment:
The discussion whether touching the works is the right cache busting strategy hasn't quite concluded, but it looks like we were leaning towards it. So, this can be optimized further.

For the gift works, the information is in the blurb. The blurb cache key depends on the work cache key, which depends on the work's updated_at. So it's enough to touch the work to bust the cache, no need to manually do so via `expire_caches`. That means that this can use the more optimised touch_all instead of looping through the gift works.

Your PR doesn't say why you updated `pseud.works` to use `expire_caches` as well, but I think it should be fine to go back to `touch_all` in this case too.